File: backend/src/AIWorkHub.Domain/Entities/Project.cs
using AIWorkHub.Domain.Enums;
using AIWorkHub.SharedKernel.Entities;

namespace AIWorkHub.Domain.Entities;

public sealed class Project : SoftDeleteEntity
{
    public string Name { get; set; } = string.Empty;

    public string? Description { get; set; }

    public string Color { get; set; } = "#2563EB";

    public ProjectStatus Status { get; set; }

    public ProjectPriority Priority { get; set; }

    public bool IsArchived { get; set; }

    public bool IsFavorite { get; set; }

    public DateTime? StartDateUtc { get; set; }

    public DateTime? TargetCompletionDateUtc { get; set; }

    public DateTime? CompletedAtUtc { get; set; }

    public decimal Progress { get; set; }

    public Guid OwnerId { get; set; }

    public User Owner { get; set; } = null!;

    public ICollection<WorkTask> Tasks { get; set; } = new List<WorkTask>();
    public ICollection<ProjectMember> Members { get; set; } = new List<ProjectMember>();

}

File: frontend/frontend/src/features/projects/types/project.ts
import type {
    ProjectPriority,
    ProjectStatus
} from "@/shared/constants/project";

export interface Project {
    id: string;
    name: string;
    description?: string;
    color: string;
    status: ProjectStatus;
    priority: ProjectPriority;
    isArchived: boolean;
    isFavorite: boolean;
    progress: number;
    startDateUtc?: string;
    targetCompletionDateUtc?: string;
    completedAtUtc?: string;
    createdAtUtc: string;

}

export interface GetProjectsParams {
    search?: string;
    status?: ProjectStatus;
    priority?: ProjectPriority;
    favorite?: boolean;
    archived?: boolean;
    page?: number;
    pageSize?: number;

}

export interface CreateProjectRequest {
    name: string;
    description?: string;
    color: string;
    priority: ProjectPriority;
    startDateUtc?: string;
    targetCompletionDateUtc?: string;
}

export interface UpdateProjectRequest
    extends CreateProjectRequest {
    status: ProjectStatus;
    isFavorite: boolean;
}
